perception/autoware_tensorrt_yolox/src/tensorrt_yolox.cpp

```diff
@@ -156,16 +156,25 @@ namespace autoware::tensorrt_yolox
 TrtYoloX::TrtYoloX(
   const std::string & model_path, const std::string & precision, const int num_class,
   const float score_threshold, const float nms_threshold, tensorrt_common::BuildConfig build_config,
-  const bool use_gpu_preprocess, std::string calibration_image_list_path, const double norm_factor,
-  [[maybe_unused]] const std::string & cache_dir, const tensorrt_common::BatchConfig & batch_config,
-  const size_t max_workspace_size, const std::string & color_map_path)
+  const bool use_gpu_preprocess, const uint8_t gpu_id, std::string calibration_image_list_path,
+  const double norm_factor, [[maybe_unused]] const std::string & cache_dir,
+  const tensorrt_common::BatchConfig & batch_config, const size_t max_workspace_size,
+  const std::string & color_map_path)
+: gpu_id_(gpu_id)
 {
+  std::cout << "GPU " << gpu_id_ << " is selected for the inference!" << std::endl;
+  if (!setCudaDeviceId(gpu_id_)) {
+    throw std::runtime_error(
```

Comment:
Can you use try-catch and `RCLCPP_ERROR` instead of throw?

cf. https://google.github.io/styleguide/cppguide.html#Exceptions

Reply:
Hello @Shin-kyoto-san :raised_hands: , since tensorrt_yolox.cpp doesn't include rclcpp and is independent of ROS, I believe we don't need to use RCLCPP loggers. What do you think? ([similar comment](https://github.com/autowarefoundation/autoware.universe/pull/8245#discussion_r1718623591))

Reply:
Hi, @ismetatabay -san !!
Thank you for your reply!

C++ exceptions should only be used when it's acceptable for the process to die if they are not caught. In other words, in situations where Autoware has no other option but to terminate. 
We want to avoid this because there's a possibility that nodes might suddenly die while in use.

@Owen-Liuyuxuan -san, what do you think about it?

Reply:
@Shin-kyoto -san & @Owen-Liuyuxuan -san, thank you for your valuable comments. I have implemented a flag and pushed the corresponding commit. I have also removed the exception mechanism. Please check it and let me know if you have any suggestions on that